src/online_providers/stripe_sca.js

```diff
@@ -1,100 +1,191 @@
+//       //             }
+//       //             if (status == 404 && data.error) {
+//       //               timeout = setTimeout(waitForHookResponse, 10000);
+//       //             }
+//       //             if (status == 200 && data.data) {
+//       //               Sentry.captureMessage('Result of validatePaymentIntent', {extra: data});
+//       //               return resolve({
+//       //                 payment_intent_id: data.data.id,
+//       //                 payment_method_id: data.data.payment_method
+//       //               });
+//       //             }
+//       //           });
+//       //       })();
+//       //   });
+//       // }
+
+//       // const confirmCardPayment = stripe.confirmCardPayment(clientSecret).then((result) => {
+//       //   Sentry.captureMessage('Result of confirmCardPayment', {extra: result});
+//       //   if (result.error) {
+//       //     return Promise.reject({
+//       //       error: result.error,
+//       //       message: result.error.message,
+//       //     });
+//       //   }
+
+//       //   return ({
+//       //     payment_intent_id: result.paymentIntent.id,
+//       //     payment_method_id: result.paymentIntent.payment_method
+//       //   });
+//       // }).catch(error => {
+//       //   return {
+//       //     error,
+//       //     message: error.message,
+//       //   };
+//       // });
+//       // const validatePaymentIntent = ensureStripeHookFetched();
+//       // paymentPromise = Promise.race([confirmCardPayment, validatePaymentIntent]);
+//       showThreeDForm(hotel_id, state_token);
+//     }
+//     return paymentPromise;
+//   }
+
+
+// }
+
+function _initializeScripts() {
+  if (window.addEventListener) {
+    window.addEventListener('message', windowEventHandler);
+  } else if (window.attachEvent) {
+    window.attachEvent('message', windowEventHandler);
+  }
+}
 
-        return ({
-          payment_intent_id: result.paymentIntent.id,
-          payment_method_id: result.paymentIntent.payment_method
-        });
-      }).catch(error => {
-        return {
-          error,
-          message: error.message,
-        };
-      });
-      const validatePaymentIntent = ensureStripeHookFetched();
-      paymentPromise = Promise.race([confirmCardPayment, validatePaymentIntent]);
+function windowEventHandler(event) {
+  var isThreeDSecure = event && event.data && event.data.hasOwnProperty('three_d_secure_data');
+
+  if (event.data) {
+    var status = event.data.success;
+    var message = event.data.message;
+
+    if (isThreeDSecure) {
+      // handle Stripe SCA 3DS event
+      const threeDSecureData = event.data;
+
+      if (status === true) {
+        console.log("IT WORKS!");
+        if (gatewaySettings.onThreeDSecureSuccess && typeof(gatewaySettings.onThreeDSecureSuccess) === 'function') {
+          gatewaySettings.onThreeDSecureSuccess(threeDSecureData);
+        }
+      } else {
+        console.log("IT DOESN'T WORK");
```

Comment:
Should we perhaps log in Sentry?

Reply:
This should be removed it is just for testing.